app/models/category.rb

```diff
@@ -142,6 +142,11 @@ def with_all_children
     return child_array.flatten
   end
 
+  def self.persons_by_listings_in_category(category)
+    return Listing::find_by_category_and_subcategory(category).map { |listing| listing.author }.uniq if category
+    return Category.all.map { |category| category.listings.map { |listing| listing.author } }.uniq
```

Comment:
I think it's better to optimize this query now, when it's being called only from one place, for example, by requesting Listing or even Author model directly.